### ISSUE TITLE
'NoneType' object has no attribute 'items' in beat if no kwargs in entry

### ISSUE THREAD

**Issue by wrvdklooster** [celery/celery#5889]
# Checklist

- [x] I have verified that the issue exists against the `master` branch of Celery.
- [x] This has already been asked to the [discussion group](https://groups.google.com/forum/#!forum/celery-users) first.
- [x] I have read the relevant section in the
  [contribution guide](http://docs.celeryproject.org/en/latest/contributing.html#other-bugs)
  on reporting bugs.
- [x] I have checked the [issues list](https://github.com/celery/celery/issues?q=is%3Aissue+label%3A%22Issue+Type%3A+Bug+Report%22+-label%3A%22Category%3A+Documentation%22)
  for similar or identical bug reports.
- [x] I have checked the [pull requests list](https://github.com/celery/celery/pulls?q=is%3Apr+label%3A%22PR+Type%3A+Bugfix%22+-label%3A%22Category%3A+Documentation%22)
  for existing proposed fixes.
- [x] I have checked the [commit log](https://github.com/celery/celery/commits/master)
  to find out if the bug was already fixed in the master branch.
- [x] I have included all related issues and possible duplicate issues
  in this issue (If there are none, check this box anyway).

## Mandatory Debugging Information

- [x] I have included the output of ``celery -A proj report`` in the issue.
    (if you are not able to do this, then at least specify the Celery
     version affected).
- [x] I have verified that the issue exists against the `master` branch of Celery.
- [x] I have included the contents of ``pip freeze`` in the issue.
- [x] I have included all the versions of all the external dependencies required
  to reproduce this bug.

## Optional Debugging Information

- [ ] I have tried reproducing the issue on more than one Python version
  and/or implementation.
- [ ] I have tried reproducing the issue on more than one message broker and/or
  result backend.
- [ ] I have tried reproducing the issue on more than one version of the message
  broker and/or result backend.
- [ ] I have tried reproducing the issue on more than one operating system.
- [ ] I have tried reproducing the issue on more than one workers pool.
- [ ] I have tried reproducing the issue with autoscaling, retries,
  ETA/Countdown & rate limits disabled.
- [ ] I have tried reproducing the issue after downgrading
  and/or upgrading Celery and its dependencies.

#### Related Issues

- https://github.com/celery/celery/issues/5627

Similar issues as was solved for this PR for args exists with kwargs.

#### Possible Duplicates

- None

## Environment & Settings

**Celery version**: 4.4.0

<details>
<summary><b><code>celery report</code> Output:</b></summary>
<p>

```
```

</p>
</details>

# Steps to Reproduce
Just run celery beat.

## Required Dependencies

* **Minimal Python Version**: N/A or Unknown
* **Minimal Celery Version**: N/A or Unknown
* **Minimal Kombu Version**: N/A or Unknown
* **Minimal Broker Version**: N/A or Unknown
* **Minimal Result Backend Version**: N/A or Unknown
* **Minimal OS and/or Kernel Version**: N/A or Unknown
* **Minimal Broker Client Version**: N/A or Unknown
* **Minimal Result Backend Client Version**: N/A or Unknown

### Python Packages

<details>
<summary><b><code>pip freeze</code> Output:</b></summary>
<p>

```
amqp==2.5.2
apipkg==1.5
apiritif==0.9.0
appdirs==1.4.3
Appium-Python-Client==0.48
appnope==0.1.0
argcomplete==1.10.0
arrow==0.10.0
asn1crypto==1.2.0
astunparse==1.6.3
atomicwrites==1.3.0
attrs==19.3.0
autopep8==1.4.4
Babel==2.7.0
backports.csv==1.0.7
bandit==1.6.2
bcrypt==3.1.7
beautifulsoup4==4.8.0
billiard==3.6.1.0
bleach==3.1.0
boto==2.49.0
bs4==0.0.1
bzt==1.14.0
cached-property==1.5.1
cachetools==3.1.1
cairocffi==1.1.0
CairoSVG==2.4.2
celery==4.4.0
certifi==2019.11.28
cffi==1.13.2
chardet==3.0.4
Click==7.0
click-completion==0.5.2
colorama==0.4.3
colorlog==4.0.2
ConfigArgParse==0.15.1
contextlib2==0.5.5
coreapi==2.3.3
coreschema==0.0.4
coverage==5.0.1
crayons==0.3.0
cryptography==2.8
cssselect==1.1.0
cssselect2==0.2.2
cssutils==1.0.2
ddt==1.2.1
decorator==4.4.1
defusedxml==0.6.0
diff-match-patch==20181111
Django==1.11.27
django-activity-stream==0.8.0
django-appconf==1.0.3
django-autofixture==0.12.1
django-celery-beat==1.5.0
django-celery-results==1.1.2
django-compressor==2.3
django-constance==2.0.0
django-countries==5.5
django-crispy-forms==1.8.1
django-debug-toolbar==2.1
django-debug-toolbar-request-history==0.1.0
django-environ==0.4.5
django-extensions==2.2.5
django-extra-fields==1.1.0
django-extra-views==0.10.0
django-fernet-fields==0.6
django-filter==2.2.0
django-fsm==2.7.0
django-guardian==1.5.1
django-haystack==2.8.1
django-import-export==1.2.0
django-jarc==1.0.15
django-jsonfield-compat==0.4.4
django-libsass==0.7
django-media-fixtures==0.0.3
django-memcache-status==2.1
django-memoize==2.2.1
django-modeltranslation==0.12.1
django-oscar==1.5.4
django-oscar-api==1.0.10.post1
django-phonenumber-field==1.3.0
django-picklefield==1.0.0
django-private-storage==2.2.1
django-ratelimit==2.0.0
django-recaptcha2==1.4.1
django-rest-hooks==1.5.0
django-rest-swagger==2.1.2
django-reversion==2.0.8
django-ses==0.8.13
django-silk==3.0.4
django-simple-history==2.8.0
django-tables2==1.16.0
django-timezone-field==3.1
django-transaction-hooks==0.2
django-treebeard==4.3
django-webtest==1.9.7
django-widget-tweaks==1.4.5
djangorestframework==3.9.4
djangorestframework-jwt==1.11.0
djangorestframework-simplejwt==4.3.0
djangorestframework-xml==1.4.0
dnspython==1.15.0
docx2txt==0.8
dparse==0.4.1
EasyProcess==0.2.8
EbookLib==0.17.1
email-validator==1.0.2
entrypoints==0.3
ephem==3.7.6.0
et-xmlfile==1.0.1
execnet==1.7.1
extract-msg==0.23.1
factory-boy==2.12.0
fake-factory==0.7.2
Faker==3.0.0
fintech==4.3.5
flake8==3.7.9
flake8-bugbear==19.8.0
flake8-isort==2.8.0
flake8-polyfill==1.0.2
Flask==1.1.1
flower==0.9.3
fpdf==1.7.2
freezegun==0.3.12
future==0.18.2
fuzzyset==0.0.19
gevent==1.5a2
geventhttpclient-wheels==1.3.1.dev2
gitdb==0.6.4
gitdb2==2.0.6
GitPython==3.0.5
google-api-python-client==1.7.11
google-auth==1.10.0
google-auth-httplib2==0.0.3
google-auth-oauthlib==0.2.0
gprof2dot==2019.11.30
graypy==2.1.0
greenlet==0.4.15
hdrpy==0.3.3
html2text==2019.9.26
html5lib==1.0.1
httplib2==0.14.0
idna==2.8
IMAPClient==2.1.0
importlib-metadata==1.3.0
inflection==0.3.1
ipaddress==1.0.23
ipcalc==1.99.0
ipython==5.3.0
ipython-genutils==0.1.0
isodate==0.6.0
isort==4.3.21
itsdangerous==1.1.0
itypes==1.1.0
jdcal==1.4.1
Jinja2==2.10.3
jsonfield==2.0.2
jsonpath-rw==1.4.0
kombu==4.6.7
libsass==0.19.4
line-profiler==2.0
locustio==0.13.5
lxml==4.4.2
M2Crypto==0.35.2
manhole==1.6.0
MarkupPy==1.14
MarkupSafe==1.1.1
mccabe==0.6.1
mock==2.0.0
mollie-api-python==1.4.5
more-itertools==8.0.2
msgpack==0.6.2
msgpack-python==0.5.6
mysqlclient==1.3.10
newrelic==5.4.0.132
nose==1.3.7
numpy==1.18.0
oauth2client==4.1.3
oauthlib==3.1.0
odfpy==1.4.0
olefile==0.46
openapi-codec==1.3.2
openpyxl==2.6.4
packaging==19.2
paramiko==2.7.1
pathlib2==2.3.5
pbr==5.4.4
pdf417gen==0.7.0
pdfminer.six==20181108
pdfrw==0.4
pep8==1.7.1
pexpect==4.2.1
phonenumbers==8.11.1
pickleshare==0.7.4
Pillow==6.2.1
pip-licenses==1.17.0
pip-review==1.0
pipdeptree==0.12.1
pkg-resources==0.0.0
pkgconfig==1.5.1
pluggy==0.13.1
ply==3.11
premailer==3.6.1
progressbar33==2.4
prompt-toolkit==1.0.13
psutil==5.6.7
psycopg2-binary==2.8.4
PTable==0.9.2
ptyprocess==0.5.1
purl==1.5
py==1.8.0
pyasn1==0.4.8
pyasn1-modules==0.2.7
pyBarcode==0.8b1
pycodestyle==2.5.0
pycountry==19.8.18
pycparser==2.19
pycryptodome==3.9.4
pycycle==0.0.8
pyflakes==2.1.1
Pygments==2.5.2
PyJWT==1.7.1
Pympler==0.8
PyNaCl==1.3.0
pyparsing==2.4.5
PyPDF2==1.26.0
Pyphen==0.9.5
pysftp==0.2.9
pytest==5.3.2
pytest-cache==1.0
pytest-cov==2.8.1
pytest-django==3.7.0
pytest-factoryboy==2.0.3
pytest-forked==1.1.3
pytest-html==1.19.0
pytest-metadata==1.7.0
pytest-mock==1.13.0
pytest-pep8==1.0.6
pytest-sftpserver==1.3.0
pytest-xdist==1.30.0
python-barcode==0.10.0
python-crontab==2.4.0
python-dateutil==2.8.1
python-gnupg==0.4.5
python-Levenshtein==0.12.0
python-memcached==1.59
python-pptx==0.6.18
python3-openid==3.1.0
python3-saml==1.9.0
pytz==2019.3
PyVirtualDisplay==0.2.5
PyYAML==5.2
pyzmq==18.1.1
qrcode==6.1
raven==6.10.0
rcssmin==1.0.6
realbrowserlocusts==0.4
requests==2.22.0
requests-mock==1.7.0
requests-oauthlib==1.3.0
requests-toolbelt==0.9.1
rest-condition==1.0.3
restnavigator==1.0.1
retrying==1.3.3
rjsmin==1.1.0
rsa==4.0
safety==1.8.5
selenium==3.141.0
sepa-generator==0.1.4
shellingham==1.3.1
simplegeneric==0.8.1
simplejson==3.17.0
six==1.12.0
smmap==0.9.0
smmap2==2.0.5
snippetinjector==1.0.0
social-auth-app-django==3.1.0
social-auth-core==3.2.0
sorl-thumbnail==12.5.0
sortedcontainers==2.1.0
soupsieve==1.9.5
SpeechRecognition==3.8.1
sqlparse==0.3.0
stevedore==1.31.0
suds-jurko==0.6
tablib==0.14.0
tabulate==0.8.6
terminaltables==3.1.0
testfixtures==6.10.3
text-unidecode==1.3
textract==1.6.3
texttable==1.6.2
tinycss==0.4
tinycss2==1.0.2
toml==0.9.2
tornado==5.1.1
tqdm==4.40.2
traitlets==4.3.2
typing==3.6.4
tzlocal==1.5.1
unicodecsv==0.14.1
Unidecode==0.4.21
uritemplate==3.0.1
urllib3==1.25.7
urwid==2.0.1
uWSGI==2.0.18
uwsgitop==0.10
vine==1.3.0
vulture==1.2
waitress==1.4.0
wcwidth==0.1.7
WeasyPrint==50
webencodings==0.5.1
WebOb==1.8.5
WebTest==2.0.33
Werkzeug==0.16.0
xlrd==1.2.0
XlsxWriter==1.2.7
xlwt==1.3.0
xmlsec==1.3.3
zeep==3.4.0
zipp==0.6.0
```

</p>
</details>

### Other Dependencies

<details>
<p>
N/A
</p>
</details>

## Minimally Reproducible Test Case

Not sure how to reproduce at this moment apart from just running celery beat.
<details>
<p>

```python
```

</p>
</details>

# Expected Behavior

Celery beat runs without errors.

# Actual Behavior

Celery beat log shows following errors:
```
Traceback (most recent call last):
  File "/home/ah/virtualenv/lib/python3.5/site-packages/celery/beat.py", line 273, in apply_entry
    result = self.apply_async(entry, producer=producer, advance=False)
  File "/home/ah/virtualenv/lib/python3.5/site-packages/celery/beat.py", line 401, in apply_async
    entry, exc=exc)), sys.exc_info()[2])
  File "/home/ah/virtualenv/lib/python3.5/site-packages/vine/five.py", line 194, in reraise
    raise value.with_traceback(tb)
  File "/home/ah/virtualenv/lib/python3.5/site-packages/celery/beat.py", line 388, in apply_async
    entry_kwargs = {k: v() if isinstance(v, BeatLazyFunc) else v for k, v in entry.kwargs.items()}
celery.beat.SchedulingError: Couldn't apply scheduled task sync_products_for_source_40023: 'NoneType' object has no attribute 'items'
```

**Comment by bob-r** [comment]
I confirm the problem. Please, reopen. The fix would be similar to https://github.com/celery/celery/pull/5631.

**Comment by auvipy** [comment]
can you come with a fix with unit test?

**Comment by auvipy** [comment]
> I confirm the problem. Please, reopen. The fix would be similar to #5631.

if you have time please check #6619
